src/integrations/watchers/indexable-post-type-change-watcher.php

```diff
@@ -0,0 +1,235 @@
+use Yoast\WP\SEO\Repositories\Indexable_Repository;
+
+use Yoast_Notification;
+use Yoast_Notification_Center;
+/**
+ * WordPress Post watcher.
+ *
+ * Fills the Indexable according to Post data.
+ */
+class Indexable_Post_Type_Change_Watcher implements Integration_Interface {
+
+	/**
+	 * The indexing helper.
+	 *
+	 * @var Indexing_Helper
+	 */
+	protected $indexing_helper;
+
+	/**
+	 * Holds the Options_Helper instance.
+	 *
+	 * @var Options_Helper
+	 */
+	private $options;
+
+	/**
+	 * Holds the Post_Type_Helper instance.
+	 *
+	 * @var Post_Type_Helper
+	 */
+	private $post_type_helper;
+
+	/**
+	 * The indexable repository.
+	 *
+	 * @var Indexable_Repository
+	 */
+	private $repository;
+
+	/**
+	 * The notifications center.
+	 *
+	 * @var Yoast_Notification_Center
+	 */
+	private $notification_center;
+
+	/**
+	 * Returns the conditionals based on which this loadable should be active.
+	 *
+	 * @return array
+	 */
+	public static function get_conditionals() {
+		return [ Not_Admin_Ajax_Conditional::class, Admin_Conditional::class, Migrations_Conditional::class ];
+	}
+
+	/**
+	 * Indexable_Post_Type_Change_Watcher constructor.
+	 *
+	 * @param Options_Helper            $options             The options helper.
+	 * @param Indexable_Repository      $repository          The Indexables repository.
+	 * @param Indexing_Helper           $indexing_helper     The indexing helper.
+	 * @param Post_Type_Helper          $post_type_helper    The post_typehelper.
+	 * @param Yoast_Notification_Center $notification_center The notification center.
+	 */
+	public function __construct(
+		Options_Helper $options,
+		Indexable_Repository $repository,
+		Indexing_Helper $indexing_helper,
+		Post_Type_Helper $post_type_helper,
+		Yoast_Notification_Center $notification_center
+	) {
+		$this->options             = $options;
+		$this->repository          = $repository;
+		$this->indexing_helper     = $indexing_helper;
+		$this->post_type_helper    = $post_type_helper;
+		$this->notification_center = $notification_center;
+	}
+
+	/**
+	 * Initializes the integration.
+	 *
+	 * This is the place to register hooks and filters.
+	 *
+	 * @return void
+	 */
+	public function register_hooks() {
+		\add_action( 'admin_init', [ $this, 'check_post_types_public_availability' ] );
+	}
+
+	/**
+	 * Checks if one or more post types change visibility
+	 *
+	 * @return void
+	 */
+	public function check_post_types_public_availability() {
+
+		// We have to make sure this is just a plain http request, no ajax/REST.
+		if ( \wp_is_json_request() ) {
+			return;
+		}
+
+		$public_post_types            = \array_keys( $this->post_type_helper->get_public_post_types() );
+		$last_known_public_post_types = $this->options->get( 'last_known_public_post_types', [] );
+
+		if ( empty( $last_known_public_post_types ) ) {
+			$this->options->set( 'last_known_public_post_types', $public_post_types );
+			$last_known_public_post_types = $public_post_types;
+			return;
```

Comment:
Are we sure the `return` is right? If so, it would be useless to set the local variable `$last_known_public_post_types`

Reply:
Mabe the whole `if` is not needed at all, looking at the code after that